chutney/ui/src/app/core/services/login.service.ts

```diff
@@ -7,134 +7,192 @@
 
 import { Injectable } from '@angular/core';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
-import { Router } from '@angular/router';
-import { BehaviorSubject, Observable } from 'rxjs';
-import { delay, tap } from 'rxjs/operators';
+import { ActivatedRouteSnapshot, Router } from '@angular/router';
+import { BehaviorSubject, firstValueFrom, Observable, timeout } from 'rxjs';
+import { catchError, delay, filter, first, tap } from 'rxjs/operators';
 
 import { environment } from '@env/environment';
 import { Authorization, User } from '@model';
 import { contains, intersection, isNullOrBlankString } from '@shared/tools';
+import { SsoService } from '@core/services/sso.service';
+import { TranslateService } from '@ngx-translate/core';
+import { AlertService } from '@shared';
 
 @Injectable({
-  providedIn: 'root'
+    providedIn: 'root'
 })
 export class LoginService {
 
-  private url = '/api/v1/user';
-  private loginUrl = this.url + '/login';
-  private NO_USER = new User('');
-  private user$: BehaviorSubject<User> = new BehaviorSubject(this.NO_USER);
-
-  constructor(
-    private http: HttpClient,
-    private router: Router
-  ) { }
-
-  initLogin(url?: string) {
```

Comment:
I don't know if there is a way to add a test for this method. It contains many functional rules